pkg/apis/hco/v1beta1/hyperconverged_webhook.go

```diff
@@ -191,3 +195,22 @@ func (a *nsMutator) InjectDecoder(d *admission.Decoder) error {
 	a.decoder = d
 	return nil
 }
+
+func (r *HyperConverged) Default() {
+	hcolog.Info("handle the HyperConverged default values")
+	if r.Spec.PermittedHostDevices == nil {
```

Comment:
This is the most delicate point in my opinion:
once this mutating webhook will inject the default values on its first execution, the values will be persisted on API server so on the next attempt it will not be nil anymore.

The question is what is going to happen on upgrades: since the value is never going to be nil after an upgrade, we are never going to add (if needed but I'd expect so) other devices on future versions.

Reply:
Agree. This is WIP, and the first thing is to check that the mechanism is working, to generate webhook and so on. I will improve the logic and the API, to handle upgrades.